zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -70,6 +70,23 @@
     private final PreparedStatement selectTraceDurationTail;
     private final Map<String,String> metadata;
 
+    private final ThreadLocal<Set<String>> writtenNames = new ThreadLocal<Set<String>>() {
+                private long hour = TimeUnit.MILLISECONDS.toHours(System.currentTimeMillis());
```

Comment:
Yeah, I think it would definitely help.  We would have something we could tune, should a problem arise.
Have fun with the flags :stuck_out_tongue_winking_eye: